File: Example/ios/NotificationService/NotificationService.m
#import "NotificationService.h"
#import <CleverTap-iOS-SDK/CleverTap.h>

/*
 Note: We have added two Notification Service target for Objective-C and Swift only for sample codes.
 You can activate/deactivate other target to use any one at a time.
 Steps to add/remove target: Click on Example target -> Build Phases -> Target Dependencies -> Click on add/remove items.
 */
@implementation NotificationService

- (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withContentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler {
  // Add CleverTap Account ID and Account token in your target .plist file
  [CleverTap setDebugLevel:2];
  NSDictionary *profile = @{
                              @"Name": @"testUserA1",
                              @"Identity": @123456,
                              @"Email": @"test@test.com"
  };
  [[CleverTap sharedInstance] profilePush:profile];
  [[CleverTap sharedInstance] recordNotificationViewedEventWithData: request.content.userInfo];

  [super didReceiveNotificationRequest:request withContentHandler:contentHandler];
}

@end

File: Example/ios/NotificationServiceSwift/NotificationService.swift
import CTNotificationService
import CleverTapSDK

/*
 Note: We have added two Notification Service target for Objective-C and Swift only for sample codes.
 You can activate/deactivate other target to use any one at a time.
 Steps to add/remove target: Click on Example target -> Build Phases -> Target Dependencies -> Click on add/remove items.
 */
class NotificationService: CTNotificationServiceExtension {
  override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
    let profile: Dictionary<String, AnyObject> = [
      "Name": "testUserA1" as AnyObject,
      "Identity": 123456 as AnyObject,
      "Email": "test@test.com" as AnyObject ]
    CleverTap.sharedInstance()?.profilePush(profile)
    CleverTap.sharedInstance()?.recordNotificationViewedEvent(withData: request.content.userInfo)
    super.didReceive(request, withContentHandler: contentHandler)
  }
}
